starfyre/file_router.py

```diff
@@ -0,0 +1,116 @@
+        self.pages_directory = pages_directory
+
+    def generate_routes(self):
+        """
+        Generate routes and create corresponding HTML files.
+
+        This method generates routes based on the file names in the specified pages directory. Each file in the
+        pages directory with a ".fyre" extension is considered a separate route. The route names are derived from
+        the file names by removing the ".fyre" extension and converting the names to lowercase.
+
+        The generated route names are stored in a list, and corresponding HTML files are created in the specified
+        "dist" directory. The HTML files are created by transpiling the components using the `_build_output` method.
+
+        Note:
+        - This method should be called after initializing the `FileRouter` object.
+        - The `_build_output` method is responsible for generating the HTML files.
+
+        Example:
+            pages_directory = "test_app/pages"
+            file_router = FileRouter(pages_directory)
+            file_router.generate_routes()  # This generates the routes and corresponding HTML files.
+
+        Raises:
+            FileNotFoundError: If the specified pages directory does not exist.
+        """
+        routes = []
+        dist_dir = Path(self.pages_directory / ".." / "dist").resolve()
+        if not dist_dir.exists():
+            dist_dir.mkdir()
+
+        # get file names in the "pages" directory
+        for file_name in os.listdir(self.pages_directory):
+            if file_name.endswith(".fyre"):
+                route_name = file_name.replace(".fyre", "").lower()
+                if route_name.lower() == 'index':
+                    raise IndexFileConflictError()
+                routes.append(route_name)
+
+        # read the contents from the generated python files
+        self._build_output(generated_routes=routes, out_dir=dist_dir)
+
+    def _build_output(self, generated_routes, out_dir):
+        """
+        Transpile the output of `render_root(component)` to route HTML files.
+
+        This method takes a list of generated routes and an output directory and transpiles the components
+        using `render_root(component)`. The resulting components are then written to corresponding HTML files
+        in the output directory.
+
+        Parameters:
+            generated_routes (list): A list of route names (strings) generated from the pages directory.
+            out_dir (str): The path to the output directory where the HTML files will be written.
+
+        Example:
+            pages_directory = "test_app/pages"
+            router = FileRouter(pages_directory)
+            router.generate_routes()  # This generates the `generated_routes` list.
+            out_directory = "test_app/dist"
+            router._build_output(generated_routes, out_directory)
+        """
+        # index.html is to be generated first since it is the entry point of the app
+        generated_routes.insert(0, 'app')
```

Comment:
we should not mutate a param that is being passed. Let us create a new copy of this array.